src/core.js

```diff
@@ -47,6 +47,13 @@ let core = {
             "lib-version": "JS:" + core.versions.libVersion,
             "api-version": core.versions.apiVersion,
         }
+
+        if(customHeaders){
+            for (const key in customHeaders) {
+                headers[key] = customHeaders[key];
+            }
```

Comment:
This also be done with spread operator instead:
```suggestion
            headers = {...headers, ...customHeaders};
```

Reply:
What happens if they contain the same keys?

Reply:
In both cases the input from customHeaders will overwrite the deafult headers. This we could prevent by checking if  `headers[key] ` exist before writing the value

Reply:
I think it might be a feature that you can overwrite the default headers. So let's keep it like this.